yspryn/hw7/caching.py

```diff
@@ -0,0 +1,32 @@
+"""
+Create decorator which would remember answers,
+and if answer was given, do not make a call to a function
+"""
+
+
+def decorator_answers(func):
+    remember_result = {}
+
+    def wrapper(n: int):
+        if n in remember_result:
+            return remember_result[n]
+        res = func(n)
+        remember_result[n] = res
+        return res
+    return wrapper
```

Comment:
try to reduce amount of returns and avoid nesting if possible:

`
def wrapper(n: int):
        if n not  in remember_result:
            remember_result[n] =  func(n)
        return remember_result[n]


`